### PR TITLE
Feature/start cypress test

### PR DESCRIPTION
**Pull request by joanafbrito**
# Description drawing

- Add Integration directory
- Add cypress to scripts
- Add `onLoad()` command to the `commands,js` file 
- Add home page tests 
     * for input field with sad and happy path
     * adding name and id
     * for display of the all the cards

Next steps:

we need to add all the home page elements to the test
We need to add the tests for the pokemon_info file
We need to add the test for the How To component / and add this component 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Refactor feature (non-breaking change that modifies existing work)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

- [ ] local host 3000
- [ ] console.log()
- [ ] dev tools
- [x] cypress testing

# Checklist:

- [x] My code follows the style guidelines of this project/linted
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: cypress/integration/home_page_spec.js [hunk 1]
```diff
@@ -50,33 +52,26 @@ describe('Home page user flow', () => {
...
-
+          .type('200')
+          .should('have.value',  '200')
+          .get('button')
+          .click()
+          .get('h2')
+          .should('be.visible')
+          .contains('try again')           
+            
+    });
```
Comment (Matt-Roden) [inline cypress/integration/home_page_spec.js:63]: great job with sad path testing here.

File: cypress/integration/home_page_spec.js [hunk 2]
```diff
@@ -50,33 +52,26 @@ describe('Home page user flow', () => {
...
+        cy.get('input[type="search"]')
+          .type('6')
+          .should('have.value',  '6')
+          .get('button')
+          .click()
+          .get('.pokemon-card')
+          .contains('charizard')
+          .get('img[src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/6.png"]')
+          .should('be.visible')
+    });
```
Comment (Matt-Roden) [inline cypress/integration/home_page_spec.js:75]: Awesome. Is this type of action where you were thinking about using a command?

File: cypress/integration/home_page_spec.js [hunk 3]
```diff
@@ -27,13 +29,13 @@ describe('Home page user flow', () => {
...
     it('Should show an error message if the user adds the wrong name', () => {
           cy.get('input[type="search"]')
-          .type('poop')
-          .should('have.value',  'poop')
-          .get('button')
-          .click()
-          .get('h2')
-          .should('be.visible')
-          .contains('try again')
+            .type('poop')
```
Comment (Matt-Roden) [inline cypress/integration/home_page_spec.js:32]: 💩

File: cypress/integration/home_page_spec.js [hunk 4]
```diff
@@ -2,12 +2,14 @@ describe('Home page user flow', () => {
...
     });
-
+// this first test is skipet because we don't have all the JSX elements yet
     it.skip('Should contain the Naviagtion bar with all elements ', () => {
         cy.get()
           .should('be.visible')
           .get()
           .contains()
+
+        //   we need to add all the elements that will contain in the home page - later also test for the animations
```
Comment (Matt-Roden) [inline cypress/integration/home_page_spec.js:12]: I like that you added the tests already for what will be in place in the future. Smart thinking! 🧐

File: cypress/support/commands.js [hunk 5]
```diff
@@ -1,7 +1,9 @@
 Cypress.Commands.add('onLoad', () => {
     const baseDataURL = 'https://pokeapi.co/api/v2/pokemon';
 
-    cy.intercept(`${ baseDataURL }?limit=151`, {
+    // got an error because our map changes the url to be able to access the image, see how can we add the image url as well 
+
+    cy.intercept(`${ baseDataURL }?limit=6`, {
         pokeDex:[
             {
```
Comment (Matt-Roden) [inline cypress/support/commands.js:8]: Great job with the command. I like that we can stub all of the data for every test before hand and really gain maximum control of our test environment and expected outcomes.

File: cypress/integration/home_page_spec.js [hunk 6]
```diff
@@ -2,12 +2,14 @@ describe('Home page user flow', () => {
     beforeEach(() => {
         cy.onLoad()
```
Comment (Matt-Roden) [inline cypress/integration/home_page_spec.js:3]: 🏄‍♀️

**Review by Matt-Roden — APPROVED** [review]
Great job. I would love to have a quick walk through on the command and how you went about making it. Seems pretty straight forward from what I am reading though.

File: cypress/integration/home_page_spec.js (hunk 1 above)
Comment (hayleyw7) [reply cypress/integration/home_page_spec.js:63]: Good thinking!

File: cypress/integration/home_page_spec.js (hunk 3 above)
Comment (hayleyw7) [reply cypress/integration/home_page_spec.js:32]: lol

**Pull request merged**